// @atlaskit/section-message/dist/cjs/components/SectionMessage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _button = _interopRequireDefault(require("@atlaskit/button"));

var _theme = require("../theme");

var _styled = require("./styled");

var SectionMessage =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(SectionMessage, _Component);

  function SectionMessage() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, SectionMessage);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(SectionMessage)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "renderAction", function (_ref) {
      var text = _ref.text,
          onClick = _ref.onClick,
          href = _ref.href,
          linkComponent = _ref.linkComponent;

      if (onClick || href) {
        return _react.default.createElement(_styled.Action, {
          key: text
        }, _react.default.createElement(_button.default, {
          appearance: "link",
          spacing: "none",
          onClick: onClick,
          href: href,
          component: linkComponent
        }, text));
      }

      return _react.default.createElement(_styled.Action, {
        key: text
      }, text);
    });
    return _this;
  }

  (0, _createClass2.default)(SectionMessage, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          children = _this$props.children,
          title = _this$props.title,
          actions = _this$props.actions,
          appearance = _this$props.appearance,
          icon = _this$props.icon,
          linkComponent = _this$props.linkComponent;
      var appearanceObj = _theme.baseAppearanceObj[appearance] || _theme.baseAppearanceObj.info;
      var Icon = icon || appearanceObj.Icon;
      return _react.default.createElement(_styled.Container, {
        backgroundColor: appearanceObj.backgroundColor
      }, _react.default.createElement(_styled.IconWrapper, null, _react.default.createElement(Icon, {
        primaryColor: appearanceObj.primaryIconColor,
        secondaryColor: appearanceObj.backgroundColor
      })), _react.default.createElement("div", null, title ? _react.default.createElement(_styled.Title, null, title) : null, children ? _react.default.createElement(_styled.Description, null, children) : null, actions && actions.length > 0 ? _react.default.createElement(_styled.Actions, null, actions.map(function (action) {
        return _this2.renderAction((0, _objectSpread2.default)({}, action, {
          linkComponent: linkComponent
        }));
      })) : null));
    }
  }]);
  return SectionMessage;
}(_react.Component);

exports.default = SectionMessage;
(0, _defineProperty2.default)(SectionMessage, "defaultProps", {
  appearance: 'info'
});
