// @atlaskit/section-message/dist/cjs/components/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IconWrapper = exports.Action = exports.Actions = exports.Description = exports.Title = exports.Container = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var Container = _styledComponents.default.section.withConfig({
  displayName: "styled__Container",
  componentId: "sc-92zysz-0"
})(["\n  display: flex;\n  border-radius: ", "px;\n  background-color: ", ";\n  padding: ", "px;\n"], _theme.borderRadius, function (_ref) {
  var backgroundColor = _ref.backgroundColor;
  return backgroundColor;
}, _theme.math.multiply(_theme.gridSize, 2));

exports.Container = Container;

var Title = _styledComponents.default.h1.withConfig({
  displayName: "styled__Title",
  componentId: "sc-92zysz-1"
})(["\n  margin: 0;\n  ", ";\n"], _theme.typography.h500);

exports.Title = Title;

var Description = _styledComponents.default.div.withConfig({
  displayName: "styled__Description",
  componentId: "sc-92zysz-2"
})(["\n  * + & {\n    margin-top: 8px;\n  }\n"]);

exports.Description = Description;

var Actions = _styledComponents.default.ul.withConfig({
  displayName: "styled__Actions",
  componentId: "sc-92zysz-3"
})(["\n  display: block;\n  padding-left: 0;\n  * + & {\n    margin-top: 8px;\n  }\n"]);

exports.Actions = Actions;

var Action = _styledComponents.default.li.withConfig({
  displayName: "styled__Action",
  componentId: "sc-92zysz-4"
})(["\n  display: inline;\n  & + &::before {\n    color: ", ";\n    content: '\0a0\0a0\xB7  ';\n    display: inline;\n    text-align: center;\n    vertical-align: middle;\n    width: ", "px;\n  }\n"], _theme.colors.N500, _theme.math.multiply(_theme.gridSize, 2)); // If the icon is not wrapped in a div with a width, and we instead use margin or
// padding, the icon is shrunk by the padding.
// Since the icons will have a consistent size, we can treat them as pre-calculated
// space.


exports.Action = Action;

var IconWrapper = _styledComponents.default.div.withConfig({
  displayName: "styled__IconWrapper",
  componentId: "sc-92zysz-5"
})(["\n  flex: 0 0 auto;\n  width: ", "px;\n  > span {\n    margin: -2px 0;\n    vertical-align: top;\n  }\n"], _theme.math.multiply(_theme.gridSize, 5));

exports.IconWrapper = IconWrapper;
