// @atlaskit/radio/dist/cjs/RadioGroup.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _Radio = _interopRequireDefault(require("./Radio"));

var RadioGroup =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(RadioGroup, _Component);

  function RadioGroup(props) {
    var _this;

    (0, _classCallCheck2.default)(this, RadioGroup);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(RadioGroup).call(this, props));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "getProp", function (key) {
      return _this.props[key] ? _this.props[key] : _this.state[key];
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "onChange", function (event) {
      _this.setState({
        checkedValue: event.currentTarget.value
      });

      if (typeof _this.props.onChange === 'function') {
        _this.props.onChange(event);
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "buildOptions", function () {
      var _this$props = _this.props,
          options = _this$props.options,
          isRequired = _this$props.isRequired,
          onInvalid = _this$props.onInvalid;

      var checkedValue = _this.getProp('checkedValue');

      if (!options.length) return null;
      return options.map(function (option, index) {
        var optionProps = (0, _objectSpread2.default)({}, option);

        if (checkedValue !== null && option.value === checkedValue) {
          optionProps = (0, _objectSpread2.default)({}, option, {
            isChecked: true
          });
        }

        return _react.default.createElement(_Radio.default, (0, _extends2.default)({}, optionProps, {
          key: index,
          onChange: _this.onChange,
          onInvalid: onInvalid,
          isRequired: isRequired
        }));
      });
    });
    _this.state = {
      checkedValue: _this.props.checkedValue !== undefined ? _this.props.checkedValue : _this.props.defaultCheckedValue
    };
    return _this;
  }

  (0, _createClass2.default)(RadioGroup, [{
    key: "render",
    value: function render() {
      var options = this.buildOptions();
      return _react.default.createElement(_react.Fragment, null, options);
    }
  }]);
  return RadioGroup;
}(_react.Component);

exports.default = RadioGroup;
(0, _defineProperty2.default)(RadioGroup, "defaultProps", {
  onChange: function onChange() {},
  options: []
});
