// @atlaskit/radio/dist/cjs/styled/Radio.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IconWrapper = exports.LabelText = exports.Label = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = require("@atlaskit/theme");

var disabledColor = (0, _theme.themed)({
  light: _theme.colors.N80,
  dark: _theme.colors.N80
});

var Label = _styledComponents.default.label.withConfig({
  displayName: "Radio__Label",
  componentId: "sc-2hlmhw-0"
})(["\n  align-items: flex-start;\n  color: ", ";\n  ", ";\n  display: flex;\n"], function (props // $FlowFixMe - theme is not found in props
) {
  return props.isDisabled ? disabledColor(props) : _theme.colors.text(props);
}, function (_ref) {
  var isDisabled = _ref.isDisabled;
  return isDisabled ? (0, _styledComponents.css)(["\n          cursor: not-allowed;\n        "]) : '';
});

exports.Label = Label;
var borderColor = (0, _theme.themed)({
  light: _theme.colors.N40,
  dark: _theme.colors.DN80
});
var focusBorder = (0, _styledComponents.css)(["\n  stroke: ", ";\n  stroke-width: 2px;\n"], (0, _theme.themed)({
  light: _theme.colors.B100,
  dark: _theme.colors.B75
}));
var invalidBorder = (0, _styledComponents.css)(["\n  stroke: ", ";\n  stroke-width: 2px;\n"], (0, _theme.themed)({
  light: _theme.colors.R300,
  dark: _theme.colors.R300
}));
var activeBorder = (0, _styledComponents.css)(["\n  stroke: currentColor;\n  stroke-width: 2px;\n"]);
var checkedBorder = (0, _styledComponents.css)(["\n  stroke: currentColor;\n  stroke-width: 2px;\n"]);
var border = (0, _styledComponents.css)(["\n  stroke: ", ";\n  stroke-width: 2px;\n"], function (_ref2) {
  var isHovered = _ref2.isHovered,
      rest = (0, _objectWithoutProperties2.default)(_ref2, ["isHovered"]);
  return isHovered ? (0, _theme.themed)({
    light: _theme.colors.N40,
    dark: _theme.colors.DN200
  })(rest) : borderColor(rest);
});

var getBorderColor = function getBorderColor(props) {
  if (props.isDisabled) return '';
  if (props.isFocused) return focusBorder;
  if (props.isActive) return activeBorder;
  if (props.isInvalid) return invalidBorder;
  if (props.isChecked) return checkedBorder;
  return border;
};

var getDotColor = function getDotColor(props) {
  var isChecked = props.isChecked,
      isDisabled = props.isDisabled,
      isActive = props.isActive,
      rest = (0, _objectWithoutProperties2.default)(props, ["isChecked", "isDisabled", "isActive"]);
  var color = (0, _theme.themed)({
    light: _theme.colors.N10,
    dark: _theme.colors.DN10
  });

  if (isDisabled && isChecked) {
    color = (0, _theme.themed)({
      light: _theme.colors.N70,
      dark: _theme.colors.DN90
    });
  } else if (isActive && isChecked && !isDisabled) {
    color = (0, _theme.themed)({
      light: _theme.colors.B400,
      dark: _theme.colors.DN10
    });
  } else if (!isChecked) {
    color = (0, _theme.themed)({
      light: 'transparent',
      dark: 'transparent'
    });
  }

  return color(rest);
};

var getCircleColor = function getCircleColor(props) {
  var isChecked = props.isChecked,
      isDisabled = props.isDisabled,
      isActive = props.isActive,
      isHovered = props.isHovered,
      isInvalid = props.isInvalid,
      rest = (0, _objectWithoutProperties2.default)(props, ["isChecked", "isDisabled", "isActive", "isHovered", "isInvalid"]); // set the default

  var color = (0, _theme.themed)({
    light: _theme.colors.N10,
    dark: _theme.colors.DN10
  });

  if (isDisabled) {
    color = (0, _theme.themed)({
      light: _theme.colors.N20,
      dark: _theme.colors.DN10
    });
  } else if (isActive) {
    color = (0, _theme.themed)({
      light: _theme.colors.B50,
      dark: _theme.colors.B200
    });
  } else if (isHovered && isChecked) {
    color = (0, _theme.themed)({
      light: _theme.colors.B300,
      dark: _theme.colors.B75
    });
  } else if (isHovered) {
    color = (0, _theme.themed)({
      light: _theme.colors.N30,
      dark: _theme.colors.DN30
    });
  } else if (isChecked) {
    color = (0, _theme.themed)({
      light: _theme.colors.B400,
      dark: isInvalid ? _theme.colors.DN10 : _theme.colors.B400
    });
  }

  return color(rest);
};

var LabelText = _styledComponents.default.div.withConfig({
  displayName: "Radio__LabelText",
  componentId: "sc-2hlmhw-1"
})(["\n  padding: 2px 4px;\n"]);

exports.LabelText = LabelText;

var IconWrapper = _styledComponents.default.span.withConfig({
  displayName: "Radio__IconWrapper",
  componentId: "sc-2hlmhw-2"
})(["\n  line-height: 0;\n  flex-shrink: 0;\n  color: ", ";\n  fill: ", ";\n  transition: all 0.2s ease-in-out;\n\n  /* This is adding a property to the inner svg, to add a border to the radio */\n  & circle:first-of-type {\n    transition: stroke 0.2s ease-in-out;\n    ", ";\n  }\n"], getCircleColor, getDotColor, getBorderColor);

exports.IconWrapper = IconWrapper;
