// @atlaskit/section-message/dist/cjs/theme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.baseAppearanceObj = void 0;

var _theme = require("@atlaskit/theme");

var _warning = _interopRequireDefault(require("@atlaskit/icon/glyph/warning"));

var _error = _interopRequireDefault(require("@atlaskit/icon/glyph/error"));

var _checkCircle = _interopRequireDefault(require("@atlaskit/icon/glyph/check-circle"));

var _info = _interopRequireDefault(require("@atlaskit/icon/glyph/info"));

var _questionCircle = _interopRequireDefault(require("@atlaskit/icon/glyph/question-circle"));

var baseAppearanceObj = {
  info: {
    backgroundColor: _theme.colors.B50,
    Icon: _info.default,
    primaryIconColor: _theme.colors.B500
  },
  warning: {
    backgroundColor: _theme.colors.Y50,
    Icon: _warning.default,
    primaryIconColor: _theme.colors.Y500
  },
  error: {
    backgroundColor: _theme.colors.R50,
    Icon: _error.default,
    primaryIconColor: _theme.colors.R500
  },
  confirmation: {
    backgroundColor: _theme.colors.G50,
    Icon: _checkCircle.default,
    primaryIconColor: _theme.colors.G500
  },
  change: {
    backgroundColor: _theme.colors.P50,
    Icon: _questionCircle.default,
    primaryIconColor: _theme.colors.P500
  }
};
exports.baseAppearanceObj = baseAppearanceObj;
